bluemira/equilibria/flux_surfaces.py

```diff
@@ -871,3 +873,102 @@ def poloidal_angle(Bp_strike, Bt_strike, gamma):
     sin_theta = num / Bp_strike
 
     return np.rad2deg(np.arcsin(sin_theta))
+
+
+def harmonic_amplitude(x_f, z_f, I_f, max_degree, r_t, cylindrical=False):
+    """
+
+    Parameters
+    ----------
+    x_f: np.array
+        X coordinates of filaments (coils)
+    z_f: np.array
+        Z coordinates of filaments (coils)
+    I_f: np.array
+        Filament currents
+    max_degree: integer
+        Maximum degree of harmonic to calculate up to
+    r_t: float
+        Typical length scale (e.g. radius at outer midplane)
+    cylindrical: boolean
+        Check to see if use of cylindrical coordinates are desired
+        # NOTE: unsure exactly how to implement
+
+    Returns
+    -------
+    amplitudes: np.array
+        Array of spherical harmonic amplitudes for each harmonic from given currents
+    """
+
+    r_f = np.sqrt(x_f**2 + z_f**2)
+    theta_f = np.arctan2(x_f, z_f)
+
+    currents2harmonics = np.zeros((max_degree, np.size(x_f)))
+    for degree in np.arange(max_degree):
+        currents2harmonics[degree, :] = (
+            0.5
+            * const.mu_0
+            * (r_t / r_f) ** degree
+            * np.sin(theta_f)
+            * lpmv(1, degree, np.cos(theta_f))
+            / np.sqrt(degree * (degree + 1))
+        )
+
+    amplitudes = np.matmul(currents2harmonics, I_f)
+
+    return amplitudes
```

Comment:
Just fyi the '@' symbol I think accomplished the same thing (uses matmul under the hood) also you could just directly return the calculation eg 
```suggestion
    return currents2harmonics @ I_f
```